File: MiniDOMOffice/Office/Uscite/Reports/RUStatItem.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Office
    {

        /// <summary>
        /// Statistiche sulle uscite di un operatore per un punto operativo
        /// </summary>
        [Serializable]
        public class RUStatItem 
            : DMD.XML.DMDBaseXMLObject, IComparable, IComparable<RUStatItem>
        {
            /// <summary>
            /// ID dell'operatore
            /// </summary>
            public int UserID;

            /// <summary>
            /// Nome dell'operatore
            /// </summary>
            public string UserName;

            /// <summary>
            /// Nome dell'operatore
            /// </summary>
            public string NomeOperatore;

            /// <summary>
            /// Collezione di uscite
            /// </summary>
            [NonSerialized] public CCollection<Uscita> Uscite;

            /// <summary>
            /// Costruttore
            /// </summary>
            /// <param name="operatore"></param>
            public RUStatItem(Sistema.CUser operatore)
            {
                this.UserID = DBUtils.GetID(operatore, 0);
                this.UserName = operatore.UserName;
                this.NomeOperatore = operatore.Nominativo;
                this.Uscite = new CCollection<Uscita>();
            }

            /// <summary>
            /// Compara due elementi in base al NomeOperatore
            /// </summary>
            /// <param name="withBlock"></param>
            /// <returns></returns>
            int CompareTo(RUStatItem withBlock)
            {
                return DMD.Strings.Compare(NomeOperatore, withBlock.NomeOperatore, true);
            }

            int IComparable.CompareTo(object obj)
            {
                return this.CompareTo((RUStatItem)obj);
            }

            protected override void SetFieldInternal(string fieldName, object fieldValue)
            {
                switch (fieldName ?? "")
                {
                    case "UserID":
                        {
                            UserID = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }

                    case "UserName":
                        {
                            UserName = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                            break;
                        }

                    case "NomeOperatore":
                        {
                            NomeOperatore = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                            break;
                        }

                    case "Uscite":
                        {
                            Uscite = (CCollection<Uscita>)DMD.XML.Utils.Serializer.ToObject(fieldValue);
                            break;
                        }
                    default:
                        base.SetFieldInternal(fieldName, fieldValue);
                        break;
                }
                
            }

            /// <summary>
            /// Serializzazione xml
            /// </summary>
            /// <param name="writer"></param>
            protected override void XMLSerialize(XMLWriter writer)
            {
                writer.WriteAttribute("UserID", UserID);
                writer.WriteAttribute("UserName", UserName);
                writer.WriteAttribute("NomeOperatore", NomeOperatore);
                base.XMLSerialize(writer);
                writer.WriteTag("Uscite", Uscite);
            }

        }
    }
}

File: MiniDOMOffice/Office/Uscite/Reports/RUStatItem.vb
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Class Office

      Public Class RUStatItem
        Implements IComparable, XML.IDMDXMLSerializable

        Public UserID As Integer
        Public UserName As String
        Public NomeOperatore As String
        Public Uscite As CCollection(Of Uscita)

        Public Sub New(ByVal operatore As CUser)
            DMDObject.IncreaseCounter(Me)
            Me.UserID = GetID(operatore)
            Me.UserName = operatore.UserName
            Me.NomeOperatore = operatore.Nominativo
            Me.Uscite = New CCollection(Of Uscita)
        End Sub


        Private Function CompareTo(obj As Object) As Integer Implements IComparable.CompareTo
            With DirectCast(obj, RUStatItem)
                Return Strings.Compare(Me.NomeOperatore, .NomeOperatore, CompareMethod.Text)
            End With
        End Function

        Private Sub SetFieldInternal(fieldName As String, fieldValue As Object) Implements XML.IDMDXMLSerializable.SetFieldInternal
            Select Case fieldName
                Case "UserID" : Me.UserID = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "UserName" : Me.UserName = XML.Utils.Serializer.DeserializeString(fieldValue)
                Case "NomeOperatore" : Me.NomeOperatore = XML.Utils.Serializer.DeserializeString(fieldValue)
                Case "Uscite" : Me.Uscite = XML.Utils.Serializer.ToObject(fieldValue)
            End Select
        End Sub

        Private Sub XMLSerialize(writer As XML.XMLWriter) Implements XML.IDMDXMLSerializable.XMLSerialize
            writer.WriteAttribute("UserID", Me.UserID)
            writer.WriteAttribute("UserName", Me.UserName)
            writer.WriteAttribute("NomeOperatore", Me.NomeOperatore)
            writer.WriteTag("Uscite", Me.Uscite)
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub
    End Class




End Class